crates/sui-framework/packages/bridge/sources/committee.move

```diff
@@ -98,18 +127,122 @@ module bridge::committee {
+        }
+    }
+
+    // This method will try to create the next committee using the registration and system state,
+    // if the total stake fails to meet the minimum required percentage, it will skip the update.
+    // This is to ensure we don't fail the end of epoch transaction.
+    public(friend) fun try_create_next_committee(
+        self: &mut BridgeCommittee,
+        system_state: &SuiSystemState,
+        min_stake_participation_percentage: u8,
+    ) {
+        let validators = sui_system::validators(system_state);
+        let active_validators = validator_set::active_validators(validators);
+        let total_member_stake = 0;
+        let i = 0;
+
+        let new_members = vec_map::empty();
+
+        while (i < vec_map::size(&self.member_registration)) {
+            // retrieve registration
+            let (_, registration) = vec_map::get_entry_by_idx(&self.member_registration, i);
+            // Find validator info from system state
+            let validator_index = find_validator(active_validators, registration.sui_address);
+            // Process registration if it's active validator
+            if (option::is_some(&validator_index)) {
+                let index = option::destroy_some(validator_index);
+                let validator = vector::borrow(active_validators, index);
+                let stake_amount = validator::stake_amount(validator);
+
+                total_member_stake = total_member_stake + stake_amount;
+                let member = CommitteeMember {
+                    sui_address: registration.sui_address,
+                    bridge_pubkey_bytes: registration.bridge_pubkey_bytes,
+                    stake_amount,
+                    http_rest_url: registration.http_rest_url,
+                    blocklisted: false,
+                };
+                vec_map::insert(&mut new_members, registration.bridge_pubkey_bytes, member)
+            };
+            i = i + 1;
+        };
+
+        // Make sure the new committee represent enough stakes
+        let stake_participation_percentage = ((total_member_stake * 100 / validator_set::total_stake(
+            validators
+        )) as u8);
+
+        // Store new committee info
+        if (stake_participation_percentage >= min_stake_participation_percentage) {
+            self.total_member_stake = total_member_stake;
+            // Clear registrations
+            self.member_registration = vec_map::empty();
+            self.members = new_members
+        }
+    }
+
+    fun find_validator(validators: &vector<Validator>, validator_address: address): Option<u64> {
+        let length = vector::length(validators);
+        let i = 0;
+        while (i < length) {
+            let v = vector::borrow(validators, i);
+            if (validator::sui_address(v) == validator_address) {
+                return option::some(i)
+            };
+            i = i + 1;
+        };
+        option::none()
+    }
```

Comment:
`sui-system` has a few similar functions that are duplicated in this file, but those are either private or friend functions. What are the concerns with opening them to public? and can we change them to friend this package? @lxfind  @emmazzz